Convert DropdownListWidget, MultiSelectDropdownListWidget to ES6 classes

andersk opened the pull request:
I did some basic testing of `DropdownListWidget` instances in organization settings. We currently have no `MultiSelectDropdownListWidget` instances to test outside of the Node tests.

`git show -w` (`--ignore-all-space`) recommended for reviewing.

Cc @aryanshridhar (#18954)

timabbott commented:
I think merging this may effectively entail giving up on integrating https://github.com/zulip/zulip/pull/19591. It's possible that's already going to be pretty hard.

andersk commented:
Rebasing #19591 onto current `main` was tedious but straightforward:

https://github.com/zulip/zulip/compare/main...andersk:rebased-19591

and from there, rebasing onto this was easy:

https://github.com/zulip/zulip/compare/main...andersk:rebased-19591-on-21935

(I haven’t attempted to verify anything about these branches beyond linters and Node tests.)

timabbott commented:
OK great -- did some additional testing myself and merged. It'd be great if you can push that update to the #19591 branch.

timabbott commented:
(I thought this was messier since I hadn't pulled it down to do the `git show -w` yet)
